src/overlays/actors/ovl_En_Horse_Game_Check/z_en_horse_game_check.c

```diff
@@ -28,63 +47,454 @@ const ActorInit En_Horse_Game_Check_InitVars = {
+    s32 pad[2];
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad2;
+    EnHorse* horse = (EnHorse*)player->rideActor;
+    s32 pad3[2];
+    EnHorseGameCheck* horseGameCheck;
+
+    if (horse == NULL) {
+        return true;
+    }
+
+    if ((this->unk_168 > 50) && !(this->unk_164 & 2)) {
+        this->unk_164 |= 2;
+    } else if ((globalCtx->interfaceCtx.unk_280 >= 8) && !(this->unk_164 & 1)) {
+        this->unk_164 |= 1;
+        horse->inRace = true;
+    } else if ((globalCtx->interfaceCtx.unk_280 >= 8) && !(this->unk_164 & 8)) {
+        EnHorse* horse = this->horse1;
+
+        horse->inRace = true;
+        horse = this->horse2;
+        horse->inRace = true;
+
+        this->unk_164 |= 8;
+        this->unk_17C = this->unk_168;
+    }
+
+    if ((this->unk_17C != -1) && ((this->unk_168 - this->unk_17C) > 10)) {
+        this->unk_17C = -1;
+        func_800DFAC8(globalCtx->cameraPtrs[CAM_ID_MAIN], 4);
+    }
+
+    this->unk_168++;
+    if ((this->unk_164 & 0x4000) && (this->unk_164 & 0x200000)) {
+        if (!(this->unk_164 & 0x10)) {
+            this->unk_164 |= 0x10;
+        }
+    }
+
+    if ((this->unk_164 & 0x800) || (this->unk_164 & 0x40000) || (this->unk_164 & 0x02000000)) {
+        if (this->unk_174 > 0) {
+            this->unk_174--;
+        } else {
+            func_808F8EB0(this, globalCtx);
+        }
+        return true;
+    }
+
+    if (gSaveContext.unk_3DE0[4] >= 0x4650) {
+        Audio_QueueSeqCmd(0x8041);
+        play_sound(NA_SE_SY_START_SHOT);
+        this->unk_164 |= 0x40000;
+        gSaveContext.unk_3DD0[4] = 6;
+        SET_RACE_FLAGS(RACE_FLAG_4)
+        this->unk_174 = 60;
+    }
+
+    if (!(this->unk_164 & 0x1000)) {
+        if (Math3D_XZBoundCheck(D_808F9BAC[0], D_808F9BAC[1], D_808F9BAC[2], D_808F9BAC[3],
+                                this->horse1->actor.world.pos.x, this->horse1->actor.world.pos.z)) {
+            this->unk_164 |= 0x1000;
+        }
+    }
+
+    if (!(this->unk_164 & 0x2000) && (this->unk_164 & 0x1000)) {
+        if (Math3D_XZBoundCheck(D_808F9BBC[0], D_808F9BBC[1], D_808F9BBC[2], D_808F9BBC[3],
+                                this->horse1->actor.world.pos.x, this->horse1->actor.world.pos.z)) {
+            this->unk_164 |= 0x2000;
+        }
+    }
+
+    if (!(this->unk_164 & 0x4000) && (this->unk_164 & 0x2000)) {
+        if (Math3D_XZBoundCheck(D_808F9BCC[0], D_808F9BCC[1], D_808F9BCC[2], D_808F9BCC[3],
+                                this->horse1->actor.world.pos.x, this->horse1->actor.world.pos.z)) {
+            this->unk_164 |= 0x4000;
+        }
+    }
+
+    horseGameCheck = (EnHorseGameCheck*)DynaPoly_GetActor(&globalCtx->colCtx, this->horse1->unk_24C);
+    if ((this->unk_164 & 0x4000) && (horseGameCheck != NULL) &&
+        (horseGameCheck->dyna.actor.id == ACTOR_EN_HORSE_GAME_CHECK) &&
+        (horseGameCheck->unk_15C == ENHORSEGAMECHECK_FF_7) && !(this->unk_164 & 0x40000)) {
```

Comment:
This was marked as resolved, but don't see any changes or comments about it?

Reply:
Because I think it looks worse, and you said it was personal preference. I'll change it if we make it a generate rule or something, but if it's up to me I like it this way.

Reply:
Sounds good. Just wanted to make sure it didn't get missed.